scripts/run_integration_tests_aarch64.sh

```diff
@@ -92,33 +95,62 @@ build_edk2() {
     checkout_repo "$EDK2_PLAT_DIR" "$EDK2_PLAT_REPO" master "8227e9e9f6a8aefbd772b40138f835121ccb2307"
     checkout_repo "$ACPICA_DIR" "$ACPICA_REPO" master "b9c69f81a05c45611c91ea9cbce8756078d76233"
 
-    pushd "$EDK2_BUILD_DIR"
-    # Build
-    make -C acpica -j `nproc`
-    source edk2/edksetup.sh
-    make -C edk2/BaseTools -j `nproc`
-    build -a AARCH64 -t GCC5 -p ArmVirtPkg/ArmVirtCloudHv.dsc -b RELEASE -n 0
-    cp Build/ArmVirtCloudHv-AARCH64/RELEASE_GCC5/FV/CLOUDHV_EFI.fd "$WORKLOADS_DIR"
-    popd
+    if [[ ! -f "$EDK2_DIR/.built.edk2" || \
+          ! -f "$EDK2_PLAT_DIR/.built.edk2-plat" || \
+          ! -f "$ACPICA_DIR/.built.acpica" ]]; then
+        pushd "$EDK2_BUILD_DIR"
+        # Build
+        make -C acpica -j `nproc`
+        source edk2/edksetup.sh
+        make -C edk2/BaseTools -j `nproc`
+        build -a AARCH64 -t GCC5 -p ArmVirtPkg/ArmVirtCloudHv.dsc -b RELEASE -n 0
+        cp Build/ArmVirtCloudHv-AARCH64/RELEASE_GCC5/FV/CLOUDHV_EFI.fd "$WORKLOADS_DIR"
+        touch "$EDK2_DIR"/.built.edk2
+        touch "$EDK2_PLAT_DIR"/.built.edk2-plat
+        touch "$ACPICA_DIR"/.built.acpica
+        popd
+    fi
 }
 
 build_spdk_nvme() {
     SPDK_DIR="$WORKLOADS_DIR/spdk"
     SPDK_REPO="https://github.com/spdk/spdk.git"
+    SPDK_DEPLOY_DIR="/usr/local/bin/spdk-nvme"
     checkout_repo "$SPDK_DIR" "$SPDK_REPO" master "f9c496b8e21a8f499df268818bf8b5d8e2b19f04"
 
-    pushd $SPDK_DIR
-    git submodule update --init
-    apt-get update
-    ./scripts/pkgdep.sh
-    ./configure --with-vfio-user
-    chmod +x /usr/local/lib/python3.8/dist-packages/ninja/data/bin/ninja
-    make -j `nproc`
-    mkdir /usr/local/bin/spdk-nvme
-    cp ./build/bin/nvmf_tgt /usr/local/bin/spdk-nvme
-    cp ./scripts/rpc.py /usr/local/bin/spdk-nvme
-    cp -r ./scripts/rpc /usr/local/bin/spdk-nvme
-    popd
+    if [ ! -f "$SPDK_DIR/.built.spdk" ]; then
+        pushd $SPDK_DIR
+        git submodule update --init
+        apt-get update
+        ./scripts/pkgdep.sh
+        ./configure --with-vfio-user
+        chmod +x /usr/local/lib/python3.8/dist-packages/ninja/data/bin/ninja
+        make -j `nproc` || exit 1
+        touch .built.spdk
+        popd
+    else
```

Comment:
If it is the first build, the files below cannot be copied.
You should remove the `else` condition and move the instructions in it outside.

Reply:
Done